docs/site/Decorators.md

```diff
@@ -132,56 +132,183 @@ accordingly or do a composition of them:
+  be found in [API document](http://apidocs.loopback.io/@loopback%2fopenapi-v3/#param.query). 
 
+  An equivalent example using the shortcut decorator would be:
+  
   ```ts
-    // example 2: decorator `@param` applied on parameter level
     class MyController {
-      @get('/')
+      @get('/Pets/{category}')
       list(
-        @param(offsetSpec) offset?: number,
-        @param(pageSizeSpec) pageSize?: number,
+        @param.path.string('category') category: string,
+        @param.query.number('pageSizes') pageSize?: number,
       ) {}
     }
   ```
 
-  In the first example, we apply multiple decorators to a single declaration.
-  The order in which the decorators are called is important because a `@param`
-  decorator must be applied after an operation decorator.
-  To learn more about TypeScript decorator composition,
-  refer to [TypeScript Decorator Documentation](https://www.typescriptlang.org/docs/handbook/decorators.html)
+  You can find the specific usage in [Writing Controller methods](controllers.md#writing-controller-methods)
 
-  Please note method level `@param` and parameter level `@param` are mutually exclusive,
-  you can not mix and apply them to the same parameter.
+  *The parameter location cookie is not supported yet, see*
+  *https://github.com/strongloop/loopback-next/issues/997*
 
-  You can also use this pattern to make it quicker to define params: `@param.${in}.${type}(${name})`
+### RequestBody Decorator
 
-  - in: one of the following values: `query`, `header`, `path`, `formData`, `body`
-  - type: one of the following values:  `string`, `number`, `boolean`, `integer`
-  - name: a `string`, name of the parameter
+  Syntax: see [API documentation](https://github.com/strongloop/loopback-next/blob/master/packages/openapi-v3/src/request-body-decorator.ts#L20-L79)
+
+  `@requestBody()` is applied to a controller method parameter to generate OpenAPI requestBody specification for it.
+  
+  *Only one parameter can be decorated by `@requestBody` per controller method.*
+
+  A typical OpenAPI requestBody spec contains properties `description`, `required`, and `content`:
+
+  ```ts
+    requestBodySpec: {
+    description: 'a user',
+    required: true,
+    content: {
+      'application/json': {...schemaSpec},
+      'application/text': {...schemaSpec},
+    },
+  }
+  ```
+
+  When the decorated argument is a custom type or a model, we recommend you define 
+  a model class with `@model` and `@property`:
+  
+  ```ts
+  import {model, property} from '@loopback/repository';
+  import {Address} from './address.model';
+
+  @model()
+  class User {
+    @property()
+    firstname: string
+    @property()
+    lastname: string
+    @property()
+    address: Address
+  }
+  ```
+  *To learn more about decorating models and the corresponding OpenAPI schema, please check*
+  *[model decorators](#model-decorators).*
+
+  Then a typical usage of `@requestBody` is specifying a parameter's type as the model class,
```

Comment:
Grammatically, this sentence is incomplete. I expect a sentence listing out the condition using 'if' before the sentence using 'then'

Reply:
I get what you're trying to say now with this section. Here's how I would word it based on the context you've given:
```
In order to use 'requestBody', the parameter type it's decorating needs to have its model decorated with '@model' and '@property'

CODE EXAMPLE

This allows type information of the model to be visible to the spec generator and '@requestBody' can finally be used on the parameter:
CODE EXAMPLE
```